components/org.wso2.carbon.identity.sso.agent.oidc/src/main/java/org/wso2/carbon/identity/sso/agent/oidc/util/CommonUtils.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest;
+import org.apache.oltu.oauth2.client.response.OAuthClientResponse;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.sso.agent.oidc.SSOAgentContextEventListener;
+import org.wso2.carbon.identity.sso.agent.oidc.bean.TokenData;
+import org.wso2.carbon.identity.sso.agent.oidc.exception.SSOAgentClientException;
+import org.wso2.carbon.identity.sso.agent.oidc.exception.SSOAgentServerException;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/**
+ * This class is used to define the utilities required in sso-agent-oidc module.
+ */
+public class CommonUtils {
+
+    private static final Map<String, TokenData> TOKEN_STORE = new HashMap<>();
+
+    private CommonUtils() {
+
+    }
+
+    public static JSONObject requestToJson(final OAuthClientRequest accessRequest) {
+
+        JSONObject obj = new JSONObject();
+        obj.append("tokenEndPoint", accessRequest.getLocationUri());
+        obj.append("request body", accessRequest.getBody());
+
+        return obj;
+    }
+
+    public static JSONObject responseToJson(final OAuthClientResponse oAuthResponse) {
+
+        JSONObject obj = new JSONObject();
+        obj.append("status-code", "200");
+        obj.append("id_token", oAuthResponse.getParam("id_token"));
+        obj.append("access_token", oAuthResponse.getParam("access_token"));
+        return obj;
+
+    }
+
+    public static boolean logout(final HttpServletRequest request, final HttpServletResponse response) {
+        // Invalidate session
+        final HttpSession session = request.getSession(false);
+        if (session != null) {
+            session.invalidate();
+        }
+
+        final Optional<Cookie> appIdCookie = getAppIdCookie(request);
+
+        if (appIdCookie.isPresent()) {
+            TOKEN_STORE.remove(appIdCookie.get().getValue());
+            appIdCookie.get().setMaxAge(0);
+            response.addCookie(appIdCookie.get());
+            return true;
+        }
+        return false;
+    }
+
+    public static void getToken(final HttpServletRequest request, final HttpServletResponse response)
+            throws OAuthProblemException, OAuthSystemException, SSOAgentServerException {
+
+        final Optional<Cookie> appIdCookie = getAppIdCookie(request);
+        final HttpSession session = request.getSession(false);
```

Comment:
As we are establishing authenticated session from this point, should invalidate existing (unauthenticated session) and create new session.

Ref: https://wso2.com/technical-reports/wso2-secure-engineering-guidelines#A222